### PR TITLE
feat: flatten docs dropdown

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Flattens `Documentation` dropdowns in top-navs, as well as `Documentation` category dropdowns in the sidebars of `/<product>` pages.

> **Note**: does not remove the Terraform `Documentation` dropdown, as it is consistent with the terraform.io navigation.

## 🤷 Why

- Removes need for `General Documentation` wording
    - For all products except Terraform, `Documentation` means `/docs`, and other content like `/api` and `/cli` are called eg `API` and `CLI`
    - For Terraform, `Documentation` is the broad category and `About the docs` refers to the lone `/terraform/docs` landing page. (Unlike every other product, there are no pages in `/terraform/docs/*`, just a landing page).
- Removes inconsistency between "nested docs" implication of top nav and `/<product>` sidebar, and the "not nested docs" structure of authored content, URLs, breadcrumbs, `Back to` links, and sidebars within categories
- Reflects dot-io navs, so that visitors coming from dot-io can more clearly see content mapping in dev-dot

## 🛠️ How

- Updates `get-nav-items` to return flat docs links, rather than a `Documentation` dropdown
- Updates `generate-product-landing-nav-items` to do the same
- Updates all `<product>.json` basePaths to be named `Docs` instead of `General Documentation`

## 📸 Design Screenshots

> **Note**: Screens are shown around the `1056px` viewport width. This is the smallest viewport width at which we show items in the top nav, rather than in the mobile menu.

https://user-images.githubusercontent.com/4624598/211880565-e18e3e18-0f45-4cdf-9be3-b2fa1077b302.mp4

## 🧪 Testing

- [ ] Visit [the preview][preview], and visit each of the product pages (except Terraform, which should be unchanged)
    - The top nav should have flat docs categories
    - The sidebar on the the `/<product>` landing should have flat docs categories

## 💭 Anything else?

We may want to tweak specific content for specific products. For example, `Vagrant Cloud` isn't present in the dot-io nav for vagrantup.com, so perhaps we might not want it in this "flattened" nav. This PR takes the stance that these issues can be addressed as follow-ups - the "flattened" nav links are the exact same ones that were in the `Documentation` dropdowns, so it seems likely we'd be fine to quickly follow-up if any of them need to be removed.

---

Removing the sidebar and top-nav nesting seems to worsen the existing issue of "duplicate nav links" on the `/<product>` landing pages (links are duplicated between the top nav, the sidebar, and the main area of content via "icon cards").

There are some short-term, low-effort suggestions we could pursue to alleviate this, but these would be intentionally separate from this PR.

<details>

<summary>Potential low-effort fix, modifying product pages</summary>

CleanShot 2022-11-04 at 08 57 57@2x

</details>

We could try to line these up if we feel the "nav redundancy" issue is a dealbreaker for this PR.

[task]: https://app.asana.com/0/1203461693655903/1203524858954780/f
[preview]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app
[/hcp]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/hcp
[/terraform]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/terraform
[/packer]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/packer
[/consul]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/consul
[/vault]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/vault
[/boundary]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/boundary
[/nomad]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/nomad
[/waypoint]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/waypoint
[/vagrant]: https://dev-portal-git-zsflatten-docs-dropdown-hashicorp.vercel.app/vagrant

### CODE REVIEW

File: src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts
Comment (kendallstrautman) [inline src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts:35]: 💭 We should update the test data too to not use `General Documentation`

File: src/components/navigation-header/components/product-page-content/utils/get-nav-items.ts
Comment (kendallstrautman) [inline src/components/navigation-header/components/product-page-content/utils/get-nav-items.ts:65]: We should add some breadcrumb notes around why terraform is an outlier.

**Review by kendallstrautman — APPROVED** [review]
LGTM pending design's approval! 👍🏻

File: src/components/navigation-header/components/product-page-content/utils/get-nav-items.ts
Comment (zchsh) [reply src/components/navigation-header/components/product-page-content/utils/get-nav-items.ts:74]: 🙌 Great call! Updated with some breadcrumbs on the "Terraform has lots of docs categories, the top nav bar has limited space" issue.

File: src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts
Comment (zchsh) [reply src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts]: I dig it - updated 👍 ✅

**Comment by braydenlove** [comment]
Mike and I are meeting Thursday morning to align on all of the quick wins IA work. We will update tickets, and provide clean designs in that time and will allow us to approve existing PRs. We appreciate the patience!

**Comment by braydenlove** [comment]
In response to your _Potential low-effort fix, modifying product pages_ comment. We love this but want more time to explore, likely move forward with something smaller and create a follow up ticket to explore what you are suggesting. We agree that the redundant content is awk and can be refined!

**Comment by braydenlove** [comment]
@zchsh yup! We are working on final designs as we speak so it is crystal clear.

**Comment by heitzke** [comment]
@zchsh agreed with @braydenlove there - it _might_ be worth combining the ticket with the resized logo but it's totally your call on how to chunk up the work. I have no issue with this work being merged in as it's already complete and then take on the additional adjustments separately

**Comment by zchsh** [comment]
@heitzke @braydenlove Cool cool - personally I'm into the iterative approach. If this PR looks good to ya'll as is, if either of you could drop an ✅ `Approve` here then I'll merge this in and we'll get it live ☺️

Stoked to get started on the resized logo layout once designs are ready 💯 🙇

**Review by heitzke — APPROVED** [review]
All looks good to me here 👍 - I'd like to continue discussion with @topfunky outside this PR to unblock this and we can talk about navigation order

**Pull request merged**
